tests/test_full_course.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import unicode_literals
+
+import os
+import tempfile
+
+import shutil
+import shlex
+
+from subprocess import check_call, CalledProcessError
+
+from olxutils import cli
+
+from unittest import TestCase
+
+
+class FullCourseTestCase(TestCase):
+    """
+    Render a full course and compare its results to expectations
+    """
+
+    SOURCE_DIR = os.path.join(os.path.dirname(__file__),
+                              'course',
+                              'source')
+    RESULT_DIR = os.path.join(os.path.dirname(__file__),
+                              'course',
+                              'result')
+
+    def setUp(self):
+        """
+        Copy the source and result files to temporary directories
+        """
+        self.tmpdir = tempfile.mkdtemp()
+        self.sourcedir = os.path.join(self.tmpdir,
+                                      'source')
+        self.resultdir = os.path.join(self.tmpdir,
+                                      'result')
+        shutil.copytree(self.SOURCE_DIR,
+                        self.sourcedir,
+                        symlinks=True)
+        shutil.copytree(self.RESULT_DIR,
+                        self.resultdir,
+                        symlinks=True)
+
+    def render_course(self, cmdline):
+        os.chdir(self.sourcedir)
```

Comment:
Oh, I see you're already doing a chdir.  My hypothesis, then, would be that maybe Mako uses some method to determine the current directory other than getcwd().  Something that gets set differently by the test environment.

Reply:
Well if that were the root cause, then I'd see the error in any `tox` run, no? That's not what happens. I see it _only_ when running `tox` in the 2.7 environment. I don't see it when running `tox` with Python 3, and I don't see it when running in 2.7 outside a test case.

Reply:
Yeah.  Plus, just realized the error happens when trying to locate an olxutils module-level template, olx_partials.xml.  So, probably nothing to do with the courseware directory structure.  This is going to be gnarly.

Reply:
Yeah, exactly. Don't dig further; I'll see if I can nail this one somehow. I had hoped you might have run across this before, but if you haven't, it's fine if just one of us does the digging.

Reply:
Found and fixed it.